M3-5598: Update stories for Breadcrumb (#8186)
linode/manager: packages/manager/src/components/Breadcrumb/Breadcrumb.stories.mdx

import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
import Breadcrumb from 'src/components/Breadcrumb';
import UserIcon from 'src/assets/icons/user.svg';

<Meta
  title="UI Elements/Breadcrumb"
  component={Breadcrumb}

>   args={{
>     pathname: '/linodes/9872893679817/test/lastcrumb',
>   }}

Reviewer: Can we set some default values here so that we don't have to click the `Set ...` button to display controls?
```suggestion
  args={{
    pathname: '/linodes/9872893679817/test/lastcrumb',
    crumbOverrides: [],
    firstAndLastOnly: false,
    labelOptions: {
      subtitle: ''
    },
    labelTitle: '',
    removeCrumbX: 0
  }}
```
<img width="1026" alt="image" src="https://user-images.githubusercontent.com/14323019/153940482-b4453fa1-f633-4c18-95da-9a4a7d8e73ef.png">

Author: I think it makes sense to have defaults set for `pathname` and `firstAndLastOnly`. The other two would change the component if something is set. If the user wants to see what that version of the component is they can click an extra button.

  argTypes={{
    pathname: {
      description: "A string representation of the path of a resource. Each crumb is seperated by a '/' character.",
    },
    crumbOverrides: {
      description: 'An array of objects that can be used to customize any crumb.',
      control: false,
    },
    labelTitle: {
      description: 'A string that can be used to set a custom title for the last crumb.',
    },
    labelOptions: {
      description: 'An object that can be used to configure the final crumb.',
      control: false,

>     },

Reviewer: Instead of setting the control for `crumbOverrides` and `labelOptions` false, what about just setting it to an empty array/empty object in the args?

Author: I don't think that will work well. It probably would cause rendering issues.

    },
    onEditHandlers: {
      description: 'An object that can be used to define functions, errors, and crumb title for an editable final crumb',
      control: false,
    },
    removeCrumbX: {
      description: 'A number indicating the position of the crumb to remove. Not zero indexed.',
    },
  }}
/>

export const BreadcrumbTemplate = (args, context) => (
  <Breadcrumb {...args} />
)

export const EditableBreadcrumbTemplate = (args, context) => {
  const [text, setText] = React.useState('Editable Text!');
  return (
  <Breadcrumb 
    labelOptions={{
      linkTo: ''
    }}
    onEditHandlers={{
      editableTextTitle: text,
      onEdit: (value) => ( Promise.resolve(setText(value)) ),
      onCancel: () => {},
    }}
    {...args} />);
}

# Breadcrumb

## Usage
- Include the current page as the last item in the breadcrumb trail.
- In the breadcrumb trail, the breadcrumb corresponding the the current page should not be a link.

<Canvas>
  <Story name="Basic">
    {BreadcrumbTemplate.bind({})}
  </Story>
</Canvas>

<ArgsTable 
  story="Basic"
  sort="requiredFirst"
  exclude={['className']}
/>

> ### Breadcrum with Subtitle

Reviewer: Breadcrumb

---

<Canvas>
  <Story 
   name="Breadcrumb with subtitle"
   args={{
     labelOptions: {
       subtitle: 'subtitle',
     }
  }}>
    {BreadcrumbTemplate.bind({})}
  </Story>
</Canvas>

> ### Breadcrum with Prefix Component

Reviewer: Breadcrumb

---

<Canvas>
  <Story
   name="Breadcrumb with prefix component"
   args={{
     labelOptions: {
       prefixComponent: (
         <UserIcon
                style={{
                  margin: '2px 0 0 0',
                  color: '#606469',
                  borderRadius: '50%',
                  width: 32,
                  height: 32,
                  animation: '$fadeIn 150ms linear forwards',
                }}
              />
       )
     }
   }}>
    {BreadcrumbTemplate.bind({})}
  </Story>
</Canvas>

> ### Breadcrum with Editable Text

Reviewer: Breadcrumb

import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
import Breadcrumb from 'src/components/Breadcrumb';
import UserIcon from 'src/assets/icons/user.svg';

<Meta
  title="UI Elements/Breadcrumb"
  component={Breadcrumb}
  args={{
    pathname: '/linodes/9872893679817/test/lastcrumb',
    firstAndLastOnly: false,
  }}
  argTypes={{
    pathname: {

>       description: "A string representation of the path of a resource. Each crumb is seperated by a '/' character.",

Reviewer 2: ```suggestion
      description: "A string representation of the path of a resource. Each crumb is separated by a `/` character.",
```

    },
    crumbOverrides: {
      description: 'An array of objects that can be used to customize any crumb.',
      control: false,
    },
    labelTitle: {
      description: 'A string that can be used to set a custom title for the last crumb.',
    },
    labelOptions: {
      description: 'An object that can be used to configure the final crumb.',
      control: false,
    },
    firstAndLastOnly: {

>       description: 'A boolean that if true will only show the first and last crumb',

Reviewer 2: ```suggestion
      description: 'A boolean that if true will only show the first and last crumb.',
```

    },
    onEditHandlers: {
      description: 'An object that can be used to define functions, errors, and crumb title for an editable final crumb',
      control: false,
    },
    removeCrumbX: {
      description: 'A number indicating the position of the crumb to remove. Not zero indexed.',
    },
  }}
/>

export const BreadcrumbTemplate = (args, context) => (
  <Breadcrumb {...args} />
)

export const EditableBreadcrumbTemplate = (args, context) => {
  const [text, setText] = React.useState('Editable Text!');
  return (
  <Breadcrumb 
    labelOptions={{
      linkTo: ''
    }}
    onEditHandlers={{
      editableTextTitle: text,
      onEdit: (value) => ( Promise.resolve(setText(value)) ),
      onCancel: () => {},
    }}
    {...args} />);
}

# Breadcrumb

## Usage

> - Include the current page as the last item in the Breadcrumb trail.

Reviewer 2: ```suggestion
- Include the current page as the last item in the breadcrumb trail.
```

> - In the Breadcrumb trail, the Breadcrumb corresponding the the current page should not be a link.

Reviewer 2: ```suggestion
- In the breadcrumb trail, the breadcrumb corresponding the the current page should not be a link.
```
